cpp/arcticdb/storage/azure/azure_storage-inl.hpp

```diff
@@ -0,0 +1,340 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#ifndef ARCTICDB_AZURE_STORAGE_H_
+#error "This should only be included by file_storage.hpp"
+#endif
+
+#include <arcticdb/util/preconditions.hpp>
+#include <arcticdb/util/pb_util.hpp>
+#include <arcticdb/log/log.hpp>
+#include <arcticdb/util/buffer_pool.hpp>
+
+#include <google/protobuf/io/zero_copy_stream_impl_lite.h>
+#include <folly/gen/Base.h>
+#include <arcticdb/storage/storage_utils.hpp>
+#include <arcticdb/storage/object_store_utils.hpp>
+#include <arcticdb/storage/storage_options.hpp>
+#include <arcticdb/entity/serialized_key.hpp>
+#include <arcticdb/util/exponential_backoff.hpp>
+#include <arcticdb/util/configs_map.hpp>
+#include <arcticdb/util/composite.hpp>
+
+#include <azure/core.hpp>
+#include <azure/storage/blobs.hpp>
+
+#include <boost/interprocess/streams/bufferstream.hpp>
+#include <folly/ThreadLocal.h>
+
+#undef GetMessage
+
+namespace arcticdb::storage::azure {
+
+namespace fg = folly::gen;
+
+namespace detail {
+
+static const size_t BATCH_SUBREQUEST_LIMIT = 1000;
+
+inline Azure::Core::Context get_context(unsigned int request_timeout){
+    Azure::Core::Context requestContext; //TODO: Maybe can be static but need to be careful with its shared_ptr and ContextSharedState
+    return requestContext.WithDeadline(std::chrono::system_clock::now() + std::chrono::milliseconds(request_timeout));
+}
+
+template<class KeyBucketizer>
+void do_write_impl(
+    Composite<KeySegmentPair>&& kvs,
+    const std::string& root_folder,
+    Azure::Storage::Blobs::BlobContainerClient& container_client,
+    KeyBucketizer&& bucketizer,
+    const Azure::Storage::Blobs::UploadBlockBlobFromOptions& upload_option,
+    unsigned int request_timeout) {
```

Comment:
Please indent the argument list differently from the method body.

Our old code doesn't do it for historical reasons.